app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,36 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @user = User.find(params[:user_id])
+    @posts = @user.posts
+  end
 
-  def show; end
+  def show
+    @post = Post.find(params[:id])
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @user = current_user
+    @post = @user.posts.new(
+      title: allowed_params[:title],
+      text: allowed_params[:text],
+      author_id: @user.id,
+      CommentsCounter: 0,
+      LikesCounter: 0
+    )
+    if @post.save
+      redirect_to action: 'show', id: @post.id
+      flash.now[:notice] = 'Post created'
+    else
+      redirect_to :back
+      flash.now[:notice] = 'An error has occurred'
+    end
+  end
```

Comment:
✅  post is created on behalf of the currently logged in user 👍🏻 

This could be refactored to

```ruby
def create
      @post = @current_user.posts.new(allowed_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
    end
```